cupy/core/include/cupy/carray.cuh

```diff
@@ -1,5 +1,20 @@
 #pragma once
 
+#if __cplusplus >= 201103
```

Comment:
This won't work for MSVC. You need something like this:
https://github.com/cupy/cupy/blob/b1b215037c29cf92a0c12811ead22730917630cf/cupy/core/include/cupy/complex/complex.h#L91

Reply:
IIRC MSVC has full C++11 support starting vs2015, but my memory could be a bit vague.

Reply:
According to https://docs.microsoft.com/en-us/cpp/overview/visual-cpp-language-conformance?view=msvc-160 you are right (with a few odd-ball exceptions that don't apply here).